src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -0,0 +1,1443 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ExtractClass;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.ExtractClass;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ExtractClass
+{
+    public class ExtractClassTests : AbstractCSharpCodeActionTest
+    {
+        private Task TestAsync(
+            string input,
+            string expected,
+            IEnumerable<(string name, bool makeAbstract)> dialogSelection = null,
+            bool sameFile = false,
+            bool expectNullMember = false,
+            TestParameters testParameters = default)
+        {
+            var service = new ExtractClassOptionsService(dialogSelection, sameFile, expectNullMember);
+
+            return TestInRegularAndScript1Async(
+                input,
+                expected,
+                parameters: testParameters.WithFixProviderData(service));
+        }
+
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+           => new CSharpExtractClassCodeRefactoringProvider((IExtractClassOptionsService)parameters.fixProviderData);
+
+        [Fact]
+        public async Task TestSingleMethod()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
```

Comment:
do yo have a test where you extract into a namespace?

Reply:
Good catch. Adding

Reply:
Added